### PR TITLE
Add codespell: workflow,

### PR DESCRIPTION
**Pull request by yarikoptic**
### Submission Checklist

Please take the following actions before submitting this pull request.

* [x] Fork your PR branch from the Toolkit "develop" branch (not main)
* [x] Test all changes with pre-commit in a local branch [#](https://goo.gle/hpc-toolkit#development). If running with `-a` codespell triggers since then `pre-commit` passes all file names and thus excludes in `.codespellrc` do not apply. Should I do anything about that?
* [ ]  Confirm that "make tests" passes all tests
* Follow Cloud HPC Toolkit Contribution guidelines [#](https://goo.gle/hpc-toolkit-contributing)

### CODE REVIEW

File: pkg/config/expression.go [hunk 1]
```diff
@@ -299,7 +299,7 @@ func (e BaseExpression) key() expressionKey {
 // do not attempt to build it by other means.
 func (e BaseExpression) AsValue() cty.Value {
 	k := e.key()
-	// we don't care if ot overrides as expressions are identical
+	// we don't care if overrides as expressions are identical
```
Comment (nick-stroud) [inline pkg/config/expression.go:302]: nit: "if it overrides"

File: .github/workflows/codespell.yml
Comment (nick-stroud) [inline .github/workflows/codespell.yml:1]: We generally tend to prefer pre-commit hooks over workflows. We also happen to run all pre-commit hooks as part of pr validation before merging. I see that you have set up both pre-commit and workflow. Could you please remove the workflow from this pr, just leaving the pre-commit hook.

**Review by nick-stroud — CHANGES_REQUESTED** [review]
Thank you for the submission and apologies that this has sit for a while.

Please make the requested changes and then un-assign yourself from the PR.

**Comment by yarikoptic** [comment]
> Please make the requested changes and then un-assign yourself from the PR.

done! also rebased and reran the `codespell -w` last commit -- there were new typos (or just new detected with newer codespell I used)

<details>
<summary>Here is the interdiff between those two showing new fixups</summary> 

```shell
❯ interdiff -U1 <(git show gh-yarikoptic/enh-codespell) <(git show enh-codespell)
diff -U1 b/community/front-end/ofe/website/ghpcfe/views/credentials.py b/community/front-end/ofe/website/ghpcfe/views/credentials.py
--- b/community/front-end/ofe/website/ghpcfe/views/credentials.py
+++ b/community/front-end/ofe/website/ghpcfe/views/credentials.py
@@ -130,3 +130,3 @@
 class CredentialViewSet(viewsets.ModelViewSet):
-    """Custom ModelViewSet for Crendential model"""
+    """Custom ModelViewSet for Credential model"""
 
diff -U1 b/community/modules/compute/gke-job-template/variables.tf b/community/modules/compute/gke-job-template/variables.tf
--- b/community/modules/compute/gke-job-template/variables.tf
+++ b/community/modules/compute/gke-job-template/variables.tf
@@ -41,3 +41,3 @@
 variable "node_pool_name" {
-  description = "A list of node pool names on which to run the job. Can be populated via `use` field."
+  description = "A list of node pool names on which to run the job. Can be populated via `use` feild."
   type        = list(string)
@@ -53,3 +53,3 @@
 variable "node_pool_name" {
-  description = "A list of node pool names on which to run the job. Can be populated via `use` feild."
+  description = "A list of node pool names on which to run the job. Can be populated via `use` field."
   type        = list(string)
diff -U1 b/community/modules/file-system/nfs-server/README.md b/community/modules/file-system/nfs-server/README.md
--- b/community/modules/file-system/nfs-server/README.md
+++ b/community/modules/file-system/nfs-server/README.md
@@ -120,3 +120,3 @@
 | <a name="input_auto_delete_disk"></a> [auto\_delete\_disk](#input\_auto\_delete\_disk) | Whether or not the nfs disk should be auto-deleted | `bool` | `false` | no |
-| <a name="input_deployment_name"></a> [deployment\_name](#input\_deployment\_name) | Name of the HPC deployment, used as name of the NFS instance if no name is specified. | `string` | n/a | yes |
+| <a name="input_deployment_name"></a> [deployment\_name](#input\_deployment\_name) | Name of the HPC deployment, used as name of the NFS instace if no name is specified. | `string` | n/a | yes |
 | <a name="input_disk_size"></a> [disk\_size](#input\_disk\_size) | Storage size gb | `number` | `"100"` | no |
@@ -125,3 +125,3 @@
 | <a name="input_auto_delete_disk"></a> [auto\_delete\_disk](#input\_auto\_delete\_disk) | Whether or not the nfs disk should be auto-deleted | `bool` | `false` | no |
-| <a name="input_deployment_name"></a> [deployment\_name](#input\_deployment\_name) | Name of the HPC deployment, used as name of the NFS instace if no name is specified. | `string` | n/a | yes |
+| <a name="input_deployment_name"></a> [deployment\_name](#input\_deployment\_name) | Name of the HPC deployment, used as name of the NFS instance if no name is specified. | `string` | n/a | yes |
 | <a name="input_disk_size"></a> [disk\_size](#input\_disk\_size) | Storage size gb | `number` | `"100"` | no |
--- b/community/modules/scripts/ramble-setup/main.tf
+++ a/community/modules/scripts/ramble-setup/main.tf
@@ -24,3 +24,3 @@
     if [ -f ${var.install_dir}/share/ramble/setup-env.sh ]; then
-          echo "** Ramble's python virtualenv (/usr/local/ramble-python) is activated. Call 'deactivate' to deactivate."
+          echo "** Ramble's python virtualenv (/usr/local/ramble-python) is actiavted. Call 'deactivate' to deactivate."
           . /usr/local/ramble-python/bin/activate
diff -U1 b/modules/README.md b/modules/README.md
--- b/modules/README.md
+++ b/modules/README.md
@@ -473,3 +473,3 @@
 `terraform output`. This can useful for displaying the IP of a login node or
-printing instructions on how to use a module, as we have in the
+priting instructions on how to use a module, as we have in the
 [monitoring dashboard module](monitoring/dashboard/README.md#Outputs).
@@ -489,3 +489,3 @@
 `terraform output`. This can useful for displaying the IP of a login node or
-priting instructions on how to use a module, as we have in the
+printing instructions on how to use a module, as we have in the
 [monitoring dashboard module](monitoring/dashboard/README.md#Outputs).
only in patch2:
--- a/cmd/root.go
+++ b/cmd/root.go
@@ -197,3 +197,3 @@
 	}
-	// Code bellow assumes that `args0` contains path to file, not a
+	// Code below assumes that `args0` contains path to file, not a
 	// executable name from PATH.
only in patch2:
--- a/tools/maintenance/maintenance.py
+++ b/tools/maintenance/maintenance.py
@@ -168,3 +168,3 @@
     parser.add_argument("-m", "--print_periodic_vms", action="store_true",
-                        help="Disply nodes that have periodic" \
+                        help="Display nodes that have periodic" \
                              " maintenance setup")
```
</details>

**Comment by rohitramu** [comment]
@yarikoptic It looks like the codespell pre-commit is failing right now in the PR-validation check:

```
codespell................................................................Failed
- hook id: codespell
- exit code: 65
```

If you run the pre-commit locally, do you get the same error?

**Comment by nick-stroud** [comment]
It looks like the pre-commit does not respect the `.codespellrc` file. 

For example here is a sample of the errors:
```
Step #2 - "pre-commit-run": community/front-end/ofe/requirements.txt:4: astroid ==> asteroid
Step #2 - "pre-commit-run": community/front-end/ofe/website/ghpcfe/static/js/bootstrap.bundle.js:2947: trough ==> through
Step #2 - "pre-commit-run": community/front-end/ofe/website/ghpcfe/static/js/bootstrap.bundle.js:3420: othwerise ==> otherwise
```

These are in `.js` and `requirements.txt` files.

I was able to reproduce this locally by running `pre-commit run codespell -a`.

@yarikoptic, reassigning to you. See if you can get these skipped files observed by the pre-commit. Un-assign when you are ready for re-review. Thanks!

**Comment by yarikoptic** [comment]
sorry about that. 
- ref for issue in codespell: https://github.com/codespell-project/codespell/issues/3196 
- rebased while discarding actual run of `codespell -w`, which I then redid, and then added "custom" skips to pre-commit config

**Pull request merged**